File: 101-symmetric-tree/symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    bool isLeftandRight(TreeNode* left, TreeNode* right) {
        if (left == nullptr and right == nullptr) return true;
        if (left == nullptr or right == nullptr) return false;
        return (left->val == right->val and isLeftandRight(left->left, right->right) and isLeftandRight(left->right, right->left));

    }
public:
    bool isSymmetric(TreeNode* root) {
        return isLeftandRight(root->left, root->right);
        
    }
};

File: 101-symmetric-tree/symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def ifsym(self, arr: List[TreeNode]) -> bool:
        print("here")
        for i in range(len(arr)//2):
            if arr[i].val != arr[len(arr)-i-1].val:
                return False
        return True

    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        q = deque([root])
        while q:
            size = len(q)
            if not self.ifsym(q):
                return False
            for i in range(size):
                cur = q.popleft()
                if cur.val != 101:
                    if cur.left:
                        q.append(cur.left)
                    else:
                        q.append(TreeNode(101))
                    if cur.right:
                        q.append(cur.right)
                    else:
                        q.append(TreeNode(101))
        return True

        